### PR TITLE
Fine-tune Distributing Events section

### PR DESCRIPTION
**Pull request by smcvb**
There's too much ambiguity within the Distributing Events section. 
It should differentiate how to deal with event distribution within a bounded context and between bounded contexts.

### CODE REVIEW

File: axon-framework/events/event-bus-and-event-store.md [hunk 1]
```diff
@@ -393,14 +393,23 @@ If no explicit `eventSerializer` is configured, events are serialized using the
...
-
-Alternatively, you can choose other components to distribute events.
-Axon provides a couple of these as [extension modules](../../extensions) for example [Spring AMQP](../../extensions/spring-amqp.md) or [Kafka](../../extensions/kafka.md).
-Note that neither of these are intended as an [Event Store](#event-store), but **only** as a message broker.
\ No newline at end of file
+To distribute events between applications, it is paramount to know whether the applications belong to the same [bounded context](../../architecture-overview/ddd-cqrs-concepts.md#bounded-context).
+If they do, they're allowed to understand each other's language.
+In other words, they should be free to understand the events altogether.
+
+As such, we can share the `EventStore`'s data source between these applications typically suffices.
```
Comment (lfgcampos) [inline axon-framework/events/event-bus-and-event-store.md:400]:
```suggestion
As such, sharing the `EventStore`'s data source between these applications typically suffices.
```

File: axon-framework/events/event-bus-and-event-store.md [hunk 2]
```diff
@@ -393,14 +393,23 @@ If no explicit `eventSerializer` is configured, events are serialized using the
...
+We may thus achieve distribution by utilizing the source itself.
+You can use both the [`EmbeddedEventStore`](#embedded-event-store) and [Axon Server](../../axon-server-introduction.md) for this.
+The former would require the applications to point to the same data source, whereas the latter would require the applications to partake in the same context.
+
+However, sharing the event store is not recommended whenever the applications do not belong to the same context.
+Instead, we should protect the boundary of the contexts, both from the event's perspective and the storage solution.
+
+To distribute events between bounded contexts, you can use Axon Server's [multi-context](../../axon-server/administration/multi-context.md) solution, for example.
+The former requires application registration to specific contexts, firstly.
```
Comment (lfgcampos) [inline axon-framework/events/event-bus-and-event-store.md:409]: not sure I liked the use of `The former` here but I will leave it to @SaraTorrey to suggest something better =)

File: axon-framework/events/event-bus-and-event-store.md [hunk 3]
```diff
@@ -393,14 +393,23 @@ If no explicit `eventSerializer` is configured, events are serialized using the
...
+Secondly, you can open a stream to another context through the `AxonServerEventStore#createStreamableMessageSourceForContext(String)` operation.
+With this source in hand, you can configure a [Streaming Processor](event-processors/streaming.md) to start reading from it.
+
+Alternatively, you can choose a message broker to distribute events.
```
Comment (lfgcampos) [inline axon-framework/events/event-bus-and-event-store.md:413]: It is not super clear to me we moved from Event Storage to message distribution... when I get to this sentence, it was not clear to me that AS also supports message distribution ootb

File: axon-framework/events/event-bus-and-event-store.md (hunk 1 above)
Comment (smcvb) [reply axon-framework/events/event-bus-and-event-store.md:400]: Grammarly actually pushed me to use the current form instead of just 'sharing' :sweat_smile:

File: axon-framework/events/event-bus-and-event-store.md (hunk 3 above)
Comment (smcvb) [reply axon-framework/events/event-bus-and-event-store.md:413]: Honestly, I am not entirely clear on how to change this comment into a workable adjustment for this pull request. The section is called "Distributing Events", which within a context is simply achieved by sharing the source.

As sharing a data source is not recommended when the applications don't belong to one another, a different approach is needed.
This is where we have roughly two options:
1. Let Axon Server allow opening a stream
2. Add a third-party piece of infrastructure to distribute events

File: axon-framework/events/event-bus-and-event-store.md (hunk 1 above)
Comment (lfgcampos) [reply axon-framework/events/event-bus-and-event-store.md:400]: hahaha ok... then, IMO, it doesn't make much sense with the `we can share ... tipically suffices` 😢

File: axon-framework/events/event-bus-and-event-store.md (hunk 3 above)
Comment (lfgcampos) [reply axon-framework/events/event-bus-and-event-store.md:413]: I will think a bit more about it and get back to you... or we can just merge this one and discuss further improvements on another issue and future PR!

File: axon-framework/events/event-bus-and-event-store.md [hunk 4]
```diff
@@ -393,14 +393,30 @@ If no explicit `eventSerializer` is configured, events are serialized using the
...
-This is the approach taken by [Axon Server](../../axon-server-introduction.md), for example.
+To distribute events between applications, it is paramount to know whether the applications belong to the same [bounded context](../../architecture-overview/ddd-cqrs-concepts.md#bounded-context).
+If they do, they're allowed to understand each other's language.
+In other words, they should be free to understand the events altogether.
 
-Such an approach will also suffice whenever the [`EmbeddedEventStore`](#embedded-event-store) is used.
-Doing so requires the correct configuration of each application instance to point to the same data source.
-Listing _all_ possible data sources is out of the scope of this guide however.
-The documentation for the chosen source should thus be read for specifics on this.
+As such, we can share the `EventStore`'s data source between these applications typically suffices.
```
Comment (abuijze) [inline axon-framework/events/event-bus-and-event-store.md:400]: Well, grammarly is the tool, not the rule. The sentence doesn't make sense as it is right now.
Grammarly probably suggests "we can share" over "sharing" because it want to avoid the passive tense. However, we should remove the "typically suffices" from the sentense.

File: axon-framework/events/event-bus-and-event-store.md [hunk 5]
```diff
@@ -393,14 +393,30 @@ If no explicit `eventSerializer` is configured, events are serialized using the
 
 ## Distributing Events
 
-The use of an `EventStore` provides a simple way of distributing events: by sharing the data source.
-This is the approach taken by [Axon Server](../../axon-server-introduction.md), for example.
+To distribute events between applications, it is paramount to know whether the applications belong to the same [bounded context](../../architecture-overview/ddd-cqrs-concepts.md#bounded-context).
```
Comment (abuijze) [inline axon-framework/events/event-bus-and-event-store.md:396]: "paramount" might be over the top (pun intended :) ) here. Why not just say "important".

File: axon-framework/events/event-bus-and-event-store.md [hunk 6]
```diff
@@ -393,14 +393,30 @@ If no explicit `eventSerializer` is configured, events are serialized using the
...
+If they do, they're allowed to understand each other's language.
+In other words, they should be free to understand the events altogether.
```
Comment (abuijze) [inline axon-framework/events/event-bus-and-event-store.md:398]: Not sure what "altogether" really adds in this context. In my opinion, they aren't "allowed to" or "free to" understand the events, but when they're in the same bounded context, they just do.

File: axon-framework/events/event-bus-and-event-store.md [hunk 7]
```diff
@@ -393,14 +393,30 @@ If no explicit `eventSerializer` is configured, events are serialized using the
...
+As such, we can share the `EventStore`'s data source between these applications typically suffices.
+We may thus achieve distribution by utilizing the source itself.
+You can use both the [`EmbeddedEventStore`](#embedded-event-store) and [Axon Server](../../axon-server-introduction.md) for this.
+The former would require the applications to point to the same data source, whereas the latter would require the applications to partake in the same context.
 
-Alternatively, you can choose other components to distribute events.
-Axon provides a couple of these as [extension modules](../../extensions) for example [Spring AMQP](../../extensions/spring-amqp.md) or [Kafka](../../extensions/kafka.md).
-Note that neither of these are intended as an [Event Store](#event-store), but **only** as a message broker.
\ No newline at end of file
+However, sharing the event store is not recommended whenever the applications do not belong to the same context.
```
Comment (abuijze) [inline axon-framework/events/event-bus-and-event-store.md:405]: hmm... we need to be _really_ careful with wording here. People may read that they shouldn't use the same installation for these apps. We should make it clear that it's just about not sharing the messaging between these components, except for some clearly defined cross-boundary messages.

File: axon-framework/events/event-bus-and-event-store.md [hunk 8]
```diff
@@ -393,14 +393,30 @@ If no explicit `eventSerializer` is configured, events are serialized using the
...
+Instead, we should protect the boundary of the contexts, both from the event's perspective and the storage solution.
+Since accessing the same source directly isn't an option, in this case, we require a different solution to share events.
+
+To distribute events between bounded contexts, you can use Axon Server's [multi-context](../../axon-server/administration/multi-context.md) solution, for example.
+The multi-context support requires application registration to specific contexts, firstly.
```
Comment (abuijze) [inline axon-framework/events/event-bus-and-event-store.md:410]: If a sentence ends with "firstly", you might as well remove it. Replace the "secondly" in the next sentence with "Then,"

**Review by abuijze — CHANGES_REQUESTED** [review]

File: axon-framework/events/event-bus-and-event-store.md (hunk 1 above)
Comment (smcvb) [reply axon-framework/events/event-bus-and-event-store.md:400]: Ha, alright, alright, I'll rephrase it to get something more sensible :-)

File: axon-framework/events/event-bus-and-event-store.md (hunk 5 above)
Comment (smcvb) [reply axon-framework/events/event-bus-and-event-store.md:396]: Sounds good to if you ask me.

File: axon-framework/events/event-bus-and-event-store.md (hunk 6 above)
Comment (smcvb) [reply axon-framework/events/event-bus-and-event-store.md:398]: Hmm, I'll adjust this to some stronger wording, then.

File: axon-framework/events/event-bus-and-event-store.md (hunk 7 above)
Comment (smcvb) [reply axon-framework/events/event-bus-and-event-store.md:405]: Good point. I'll change the wording and request a rereview.

**Review by abuijze — APPROVED** [review]

**Review by lfgcampos — APPROVED** [review]
